src/test/CharterManager-unit.t.sol

```diff
@@ -441,6 +417,42 @@ contract CharterManagerTest is TestBase {
         b.frob(100 * 1e18, 50 * 1e18);
     }
 
+    function test_drip_withdraw() public {
+        init_ilk_ungate(10 * NIB_ONE_PCT);
+        (Usr a,) = init_user();
+
+        a.join(20 * 1e6);
+        a.frob(20 * 1e18, 20 * 1e18);
+        (uint256 ink, uint256 art) = a.urn();
+        assertEq(ink, 20 * 1e18);
+        assertEq(art, 20 * 1e18);
+        assertEq(a.dai(), 18 * 1e45);
+        assertEq(a.gems(), 0);
+        assertEq(vat.dai(address(vow)), 2 * 1e45);
+
+        // increase rate by 2%
+        vat.fold(ilk, address(vow), 0.02 * 1e27);
+
+        // frob out some of the funds
+        a.frob(-15 * 1e18, -15 * 1e18);
+        (ink, art) = a.urn();
+        assertEq(ink, 5 * 1e18);
+        assertEq(art, 5 * 1e18);
+        assertEq(a.gems(), 15 * 1e18);
```

Comment:
This should be checking the `a.dai()` removed by the above `a.frob()` against the 2% in stability fees collected. `-15 * 1e18` should be taking 15.3 DAI (15 + 2%) from their balance.

Reply:
Done